src/unified_message_model/message.py

```
@@ -9,28 +9,32 @@ def __init__(
         Args:
             body: The message content, as a string or bytes.
             headers: Optional dictionary of metadata associated with the message.
             message_id: Optional unique identifier for the message. If not provided, a UUID4 string is generated.
             timestamp: Optional datetime for when the message was created. Defaults to the current UTC time.
+            delay: Optional delay in milliseconds for the message.
         """
         self.id: str = message_id if message_id is not None else str(uuid.uuid4())
         self.body: Union[str, bytes] = body
         self.headers: Dict[str, Any] = headers if headers is not None else {}
         self.timestamp: datetime = timestamp if timestamp is not None else datetime.now(timezone.utc)
+        self.delay: Optional[int] = delay
```

Ruff (0.11.9)

Undefined name `Optional`

(F821)